template/sample/template.gen.go

```diff
@@ -342,106 +311,83 @@ var (
+				if err != nil {
+					return nil, fmt.Errorf("failed to eval BoolPrimitive for instance '%s': %v", instName, err)
+				}
 
-					if err != nil {
-						result = multierror.Append(result, fmt.Errorf("failed to eval DoublePrimitive for instance '%s': %v", name, err))
-						continue
-					}
+				DoublePrimitive, err := mapper.Eval(castedInst.DoublePrimitive, attrs)
 
-					StringPrimitive, err := mapper.Eval(md.StringPrimitive, attrs)
+				if err != nil {
+					return nil, fmt.Errorf("failed to eval DoublePrimitive for instance '%s': %v", instName, err)
+				}
 
-					if err != nil {
-						result = multierror.Append(result, fmt.Errorf("failed to eval StringPrimitive for instance '%s': %v", name, err))
-						continue
-					}
+				StringPrimitive, err := mapper.Eval(castedInst.StringPrimitive, attrs)
 
-					Int64Map, err := template.EvalAll(md.Int64Map, attrs, mapper)
+				if err != nil {
+					return nil, fmt.Errorf("failed to eval StringPrimitive for instance '%s': %v", instName, err)
+				}
 
-					if err != nil {
-						result = multierror.Append(result, fmt.Errorf("failed to eval Int64Map for instance '%s': %v", name, err))
-						continue
-					}
+				Int64Map, err := template.EvalAll(castedInst.Int64Map, attrs, mapper)
 
-					instances = append(instances, &istio_mixer_adapter_sample_report.Instance{
-						Name: name,
+				if err != nil {
+					return nil, fmt.Errorf("failed to eval Int64Map for instance '%s': %v", instName, err)
+				}
 
-						Value: Value,
+				_ = castedInst
 
-						Dimensions: Dimensions,
+				return &istio_mixer_adapter_sample_report.Instance{
+					Name: instName,
 
-						Int64Primitive: Int64Primitive.(int64),
+					Value: Value,
 
-						BoolPrimitive: BoolPrimitive.(bool),
+					Dimensions: Dimensions,
 
-						DoublePrimitive: DoublePrimitive.(float64),
+					Int64Primitive: Int64Primitive.(int64),
 
-						StringPrimitive: StringPrimitive.(string),
+					BoolPrimitive: BoolPrimitive.(bool),
 
-						Int64Map: func(m map[string]interface{}) map[string]int64 {
-							res := make(map[string]int64, len(m))
-							for k, v := range m {
-								res[k] = v.(int64)
-							}
-							return res
-						}(Int64Map),
-					})
-					_ = md
-				}
+					DoublePrimitive: DoublePrimitive.(float64),
 
-				if err := handler.(istio_mixer_adapter_sample_report.Handler).HandleSample(instances); err != nil {
-					result = multierror.Append(result, fmt.Errorf("failed to report all values: %v", err))
-				}
+					StringPrimitive: StringPrimitive.(string),
 
-				err := result.ErrorOrNil()
-				if err != nil {
-					return status.WithError(err)
-				}
+					Int64Map: func(m map[string]interface{}) map[string]int64 {
+						res := make(map[string]int64, len(m))
+						for k, v := range m {
+							res[k] = v.(int64)
+						}
+						return res
+					}(Int64Map),
+				}, nil
+			},
 
-				return status.OK
+			DispatchReport: func(ctx context.Context, insts interface{}, handler adapter.Handler) (adapter.ReportResult, error) {
```

Comment:
I am rethinking Eval / Process split now :-(, sorry !

1. It has the above problem
2. From the caller perspective, the response to an error from expression evaluation, and something from within the adapter is not different. We will log and return error.
3. The operator response will depend on reading the contents of the error message.